components/dada-lang/src/test_harness.rs

```diff
@@ -516,8 +516,10 @@ enum QueryKind {
 fn expected_diagnostics(path: &Path) -> eyre::Result<Vec<ExpectedDiagnostic>> {
     let file_contents = std::fs::read_to_string(path)?;
 
-    let re =
-        regex::Regex::new(r"^(?P<prefix>[^#]*)#!\s*(?P<severity>[^\s*]+)\s*(?P<msg>.*)").unwrap();
+    let re = regex::Regex::new(
+        r"^(?P<prefix>[^#]*)#!\s*(?P<highlight>\^+)?\s*(?P<severity>[^\s*]+)\s*(?P<msg>.*)",
```

Comment:
I think, @XFFXFF, that we are going to want to have a few options here, kind of like with the `#?` annotations. I think it should probably be:

* `#! ERROR` -- error that starts on this line, useful for multi-line messages
* `#! ^^^^ ERROR` -- error with complete span information

Reply:
@nikomatsakis I'm not sure I understand what you meant, actually, both `#! ERROR` and `#! ^^^^ ERROR` are already  supported, `(?P<highlight>\^+)?` will optionally capture `^`, for `#! ERROR`, `c.name("highlight")` will be `None`

Reply:
ah, great, I didn't read that closely, clearly